homeassistant/components/notion/__init__.py

```diff
@@ -0,0 +1,272 @@
+            config_entry, component)
+
+    return True
+
+
+class Notion:
+    """Define a class to handle the Notion API."""
+
+    def __init__(self, client):
+        """Initialize."""
+        self._client = client
+        self.bridges = []
+        self.sensors = []
+        self.systems = []
+        self.tasks = []
+
+    async def async_update(self, *, initialize=False):
+        """Get the latest Notion data."""
+        from aionotion.errors import NotionError
+
+        tasks = {
+            'bridges': self._client.bridge.async_all(),
+            'sensors': self._client.sensor.async_all(),
+            'tasks': self._client.task.async_all(),
+        }
+        if initialize:
+            tasks['systems'] = self._client.system.async_all()
+
+        results = await asyncio.gather(*tasks.values(), return_exceptions=True)
+        for attr, result in zip(tasks, results):
+            if isinstance(result, NotionError):
+                _LOGGER.error(
+                    'There was an error while updating %s: %s', attr, result)
+                continue
+
+            setattr(self, attr, result)
+
+
+class NotionEntity(Entity):
+    """Define a base Notion entity."""
+
+    def __init__(
+            self,
+            notion,
+            task,
+            sensor,
+            bridge,
+            system,
+            name,
+            device_class):
+        """Initialize the entity."""
+        self._async_unsub_dispatcher_connect = None
+        self._bridge = bridge
+        self._device_class = device_class
+        self._name = name
+        self._notion = notion
+        self._sensor = sensor
+        self._state = None
+        self._system = system
+        self._task = task
+
+        self._attrs = {
+            ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION,
+            ATTR_BRIDGE_MODE: self._bridge['mode'],
+            ATTR_BRIDGE_NAME: self._bridge['name'],
+            ATTR_SYSTEM_MODE: self._system['mode'],
+            ATTR_SYSTEM_NAME: self._system['name'],
+        }
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._task['id'] in [t['id'] for t in self._notion.tasks]
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return self._device_class
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Return the state attributes."""
+        return self._attrs
+
+    @property
+    def device_info(self):
+        """Return device registry information for this entity."""
+        return {
+            'identifiers': {
+                (DOMAIN, self._sensor['hardware_id'])
+            },
+            'manufacturer': 'Notion',
+            'model': self._sensor['hardware_revision'],
+            'name': self._sensor['name'],
+            'sw_version': self._sensor['firmware_version'],
+            'via_device': (DOMAIN, self._bridge['hardware_id'])
```

Comment:
Where is the bridge device registered? 

https://developers.home-assistant.io/docs/en/device_registry_index.html#defining-devices

Reply:
(you can register it inside async_setup_entry)

Reply:
Oh wow, I never knew you actually had to register the device – I thought that merely defining a `device_info` property would do everything needed. 😆 Great to know!

Reply:
https://github.com/home-assistant/home-assistant/pull/24634/commits/bae4f1895bd32371374eff3e37a2822dda8608f0